### ISSUE TITLE
Part Design: Subtractive helix doesn't subtract if number of turns exceeds the height of the object (same pitch) [latest edge]

### ISSUE THREAD

**Issue by petrubecheru** [FreeCAD/FreeCAD#14852]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

on this simple example ( 53 mm diam circle, padded to 30mm),  Subtractive helix of a simple triangle sketch will not work on 4 turns (with weird surface appearing), but works on 3 turns (same pitch). Project attached below. 

- [X] 3 turns works:
![image](https://github.com/FreeCAD/FreeCAD/assets/124100239/1da38821-c442-495b-824d-02100a6fb8fe)
![image](https://github.com/FreeCAD/FreeCAD/assets/124100239/11b3a083-973d-460e-a1e1-42ee9f2e8715)

- [ ] 4 turns doesn't work:
![image](https://github.com/FreeCAD/FreeCAD/assets/124100239/75e232b4-986e-4009-b16f-91ff2d2a09ae)
![image](https://github.com/FreeCAD/FreeCAD/assets/124100239/4919ac87-beba-4ed0-8574-8992bbe0d37b)
weird surface :
![image](https://github.com/FreeCAD/FreeCAD/assets/124100239/7f14abda-0115-45df-aa51-09d28df4a070)

### Full version info

```shell
OS: Ubuntu Core 22 (KDE/plasma)
Word size of FreeCAD: 64-bit
Version: 0.22.0dev.37843 (Git) Snap 1058
Build type: Release
Branch: main
Hash: 932c73f0de2d1e9ba3f60ae14d199cfe77255c3a
Python 3.10.12, Qt 5.15.10, Coin 4.0.0, Vtk 7.1.1, OCC 7.7.1
Locale: English/United States (en_US)
```

### Subproject(s) affected?

PartDesign

### Freecad project
Freecad [project5-4turns.zip](https://github.com/user-attachments/files/15936526/project5-4turns.zip)

### Anything else?

issue probably related to: #14851 and #14836

issue not present in stable version(0.21.2)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by maxwxyz** [comment]
Is this a duplicate: https://github.com/FreeCAD/FreeCAD/issues/14309

**Comment by andrewchambers** [comment]
Is there an OCC issue if this bug is there? Maybe good to link to it if it exists.

**Comment by chennes** [comment]
Confirming that this is fixed by #16009
